fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/patient/ChunkReader.java

```diff
@@ -48,34 +50,43 @@
  */
 public class ChunkReader extends AbstractItemReader {
     private final static Logger logger = Logger.getLogger(ChunkReader.class.getName());
-    int pageNum = 1;
-    int indexOfCurrentResourceType = 0;
+    protected int pageNum = 1;
+    protected int indexOfCurrentResourceType = 0;
     // Control the number of records to read in each "item".
-    int pageSize = Constants.DEFAULT_SEARCH_PAGE_SIZE;
+    protected int pageSize = Constants.DEFAULT_SEARCH_PAGE_SIZE;
 
-    private FHIRPersistence fhirPersistence;
-    private List<String> resourceTypes;
+    protected FHIRPersistence fhirPersistence;
+    protected List<String> resourceTypes;
+
+    // Search parameters for resource types gotten from fhir.typeFilters job parameter.
+    Map<Class<? extends Resource>, List<Map<String, List<String>>>> searchParametersForResoureTypes = null;
+    // Used to prevent the same resource from being exported multiple times when multiple _typeFilter for the same
+    // resource type are used, which leads to multiple search requests which can have overlaps of resources.
+    HashSet<String> loadedResourceIds = new HashSet<>();
```

Comment:
Is there a specific HashSet method you are using? or best to make use of Set

Reply:
changing the declaration to Set (interface) works, but what I really want is just a Hashset, not treeset, linkhashset etc, so I thought not need to use interface for the declaration.

Reply:
done